Sources/ArcGISToolkit/Components/Augmented Reality/WorldScaleGeoTrackingSceneView.swift

```diff
@@ -0,0 +1,298 @@
+            Task { await locationDataSource.stop() }
+        }
+        .task {
+            do {
+                try await locationDataSource.start()
+                await withTaskGroup(of: Void.self) { group in
+                    group.addTask {
+                        for await location in locationDataSource.locations {
+                            self.lastLocationTimestamp = location.timestamp
+                            self.currentLocation = location
+                            if let heading = currentHeading {
+                                await updateSceneView(for: location, heading: heading)
+                            }
+                        }
+                    }
+                    group.addTask {
+                        for await heading in locationDataSource.headings {
+                            self.currentHeading = heading
+                            if let location = currentLocation {
+                                await updateSceneView(for: location, heading: heading)
+                            }
+                        }
+                    }
+                }
+            } catch {}
+        }
+        .toolbar {
+            ToolbarItem(placement: .bottomBar) {
+                if !calibrationViewIsHidden {
+                    calibrationView
+                }
+            }
+        }
+        .overlay(alignment: .top) {
+            accuracyView
+                .multilineTextAlignment(.center)
+                .frame(maxWidth: .infinity, alignment: .center)
+                .padding(8)
+                .background(.regularMaterial, ignoresSafeAreaEdges: .horizontal)
+        }
+    }
+    
+    /// If necessary, updates the scene view's camera controller for a new location coming
+    /// from the location datasource.
+    @MainActor
+    private func updateSceneView(for location: Location, heading: Double) {
+        // Do not use cached location more than 10 seconds old.
+        guard abs(lastLocationTimestamp?.timeIntervalSinceNow ?? 0) < 10 else { return }
+        
+        // Make sure that horizontal and vertical accuracy are valid.
+        guard location.horizontalAccuracy > validAccuracyThreshold,
+              location.verticalAccuracy > validAccuracyThreshold else { return }
+        
+        // Make sure either the initial camera is not set, or we need to update the camera.
+        guard !initialCameraIsSet || shouldUpdateCamera(for: location) else { return }
+        
+        // Add some of the vertical accuracy to the z value of the position, that way if the
+        // GPS location is not accurate, we won't end up below the earth's surface.
+        let altitude = (location.position.z ?? 0) + location.verticalAccuracy
+        
+        cameraController.originCamera = Camera(
+            latitude: location.position.y,
+            longitude: location.position.x,
+            altitude: altitude,
+            heading: calibrationHeading ?? heading,
```

Comment:
We may adjust the heading logic here when the calibration view is implemented.

- Currently the `updateSceneView` method is called whenever a location or heading update comes from the location data source
- If the heading is manually adjusted by the user, it is no longer "connected" to the data source, and stay as the custom value forever
- If the heading is not adjusted, it will be connected to the heading updates from the LDS

I think for the `initial` mode, having the heading fixed after receiving the first update, and let the user to adjust make sense. However, for `continuous` mode, we should discuss what is the desired behavior.